internal/archiver/archiver.go

```diff
@@ -0,0 +1,73 @@
+package archiver
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/mholt/archiver/v3"
+)
+
+type FileMap struct {
+	Source string
+	Target string
+}
+
+func CreateTgz(archivePath string, workDir string, fileMaps []FileMap) error {
+	f, err := os.OpenFile(archivePath, os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return errs.Wrap(err, "Could not create temp file")
+	}
+	defer f.Close()
+	tgz := archiver.NewTarGz()
+	if err := tgz.Create(f); err != nil {
+		return errs.Wrap(err, "Could not create tar.gz")
+	}
+	defer tgz.Close()
+
+	for _, fileMap := range fileMaps {
+		source := fileMap.Source
+		if !filepath.IsAbs(source) {
+			source = filepath.Join(workDir, source)
```

Comment:
From reading the code in this file it's not clear to me why we're appending the source path to the `workDir` rather than just using the absolute path here.

Reply:
I don't follow, this is what is making the path absolute. Are you seeing another absolute path value that I can use?

Reply:
Ahh okay, maybe a comment on the `FileMap` struct would help here? I was expecting to see a call to `filepath.Abs` if the path isn't absolute but if that's how the file map is constructed it makes sense as well.

Reply:
`filepath.Abs()` I believe works with the working directory, which is a global so something we want to avoid. I'll add a comment to clarify. Fwiw the reason the paths are relative is to facilitate archiving, otherwise the archives also have absolute paths in them (ie. rather than having setup.py at the root of the archive it'd be at `c:\Users\Runner~1\` etc. (oddly specific example that totally does not indicate I struggled with this..).